#! /usr/bin/env python

PACKAGE = 'jsk_perception'
ID = "TemplateMatchDetector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

sort_direction = gen.enum([
    gen.const("NoOperation", int_t, 0, "not apply sort"),
    gen.const("Horizontal", int_t, 1, "sort by horizontal"),
    gen.const("Vertical", int_t, 2, "sort by vertical")],
                          "Sort Direction")

gen.add("template_filename", str_t, 0, "template image filename", "template.png")
gen.add("min_scale", double_t, 0, "min_scale of template", 0.4, 0.0, 1.0)
gen.add("max_scale", double_t, 0, "max_scale of template", 1.6, 1.0, 10.0)
gen.add("resize_template_num", int_t, 0, "the number of resizing template", 20, 5, 50)
gen.add("matching_threshold", double_t, 0, "matching_threshold", 0.7, 0.0, 1.0)
gen.add("nms_threshold", double_t, 0, "nms threshold", 0.5, 0.0, 1.0)
gen.add("target_num", int_t, 0, "the number of target", 6, 1, 10)
gen.add("sort_direction", int_t, 0, "Specify Sorting direction", 0, 0, 2, edit_method=sort_direction)
gen.add("update_matching_threshold", bool_t, 0, "Update matching threshold", False)
gen.add("update_step_of_threshold", double_t, 0, "Update step of matching threshold", 0.05, 0.0, 1.0)
gen.add("check_flipped_image", bool_t, 0, "Template matching using original template image and use flipped template image", True)

exit(gen.generate(PACKAGE, PACKAGE, ID))
